src/NHibernate/Impl/FilterImpl.cs

```diff
@@ -93,10 +93,19 @@ public IFilter SetParameterList(string name, ICollection values)
 			if (values.Count > 0)
 			{
 				var e = values.GetEnumerator();
-				e.MoveNext();
-				if (!type.ReturnedClass.IsInstanceOfType(e.Current))
+				try
```

Comment:
This change is done. If it is ok for you, I will merge, and add another Jira for doing same on other SetParameterList currently not generic. (And maybe on all other non generic case.)